packages/storage/src/adapters/FileSystemStorage/FileSystemStorage.ts

```diff
@@ -30,6 +30,9 @@ export class FileSystemStorage
     )
     const nodeBuffer = await file.arrayBuffer()
 
+    // Create directory if it doesn't exist
+    await fs.mkdir(path.dirname(location), { recursive: true })
```

Comment:
This is using `fs` imported from `node:fs/promises`, so that's also what we should mock. I updated the test.